src/app/providers/provider-form/github/github.component.ts

```diff
@@ -11,13 +14,24 @@ export class GithubComponent {
   providerForm: FormGroup;
 
   constructor(
+    public gitHubService: GitHubService,
     private formBuilder: FormBuilder,
     private providersService: ProvidersService) {
     this.providerForm = this.formBuilder.group(new GithubFormModel());
   }
 
   onSubmit() {
-    console.warn('Not implemented');
-    this.providersService.triggerCloseForm(this.providerForm);
+    const providerData: MainImport = {
+      url: this.providerForm.value.url,
+      token: this.providerForm.value.token,
+      group: this.providerForm.value.group,
+      project: this.providerForm.value.project,
+    };
+    this.gitHubService.githubPost(providerData)
+      .pipe(first())
+      .subscribe(
+        success => this.providersService.triggerCloseForm(this.providerForm),
+        error => {/* Error gets show to screen form window does not close. */},
```

Comment:
I don't understand what this comment means. Can you rephrase it?

Anyway, I don't think you need this empty func, you can just remove this error callback if it's ignored anyway.

Reply:
I removed it. 
The comment should say something along the lines of- when the post for the github provider completes successfully the form is automatically cleared and closed. When the post to the github provider does not complete sucessfully then the form is not closed and all the field values are retained.